test/compiler/AbstractInterpreter.jl

```diff
@@ -351,3 +351,7 @@ let NoinlineModule = Module()
         @test count(iscall((src, inlined_usually)), src.code) == 0
     end
 end
+
+# Issue #46839
+@test code_typed(()->invoke(BitSet, Any, x), ())[1][2] === Union{}
+@test code_typed(()->invoke(BitSet, Union{Tuple{Int32},Tuple{Int64}}, 1), ())[1][2] === Union{}
```

Comment:
This seems incorrect:
```
julia> invoke(BitSet, Tuple{Union{Int32,Int64}}, 1)
BitSet with 1 element:
  1
```

Reply:
When forming `Type`, it is usually required to preserve the type kind of the parameter, so that might be okay? Otherwise I suppose we need to compute the switched version of the Tuple/Union

Reply:
I'm not entirely sure what you're saying. My comment was purely saying that you cannot return `Union{}` here, because the function that it models does return.

Reply:
this exact call does fail though
```
julia> invoke(BitSet, Union{Tuple{Int32},Tuple{Int64}}, 1)
ERROR: TypeError: in invoke, expected Type{T} where T<:Tuple, got Type{Union{Tuple{Int32}, Tuple{Int64}}}
Stacktrace:
 [1] top-level scope
   @ REPL[2]:1
```

Reply:
Ok, I confused myself. Carry on.

Reply:
We perhaps should add handling for the Union case, since it is probably possible to get ml_matches for each, then Union and combine duplicate Methods. And it makes things more complete and uniform.